flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/FlinkRuleSets.scala

```diff
@@ -136,6 +136,11 @@ object FlinkRuleSets {
     FlinkLogicalNativeTableScan.CONVERTER
   )
 
+  val FLINK_LOGICAL_OPT_RULES: RuleSet = RuleSets.ofList(
```

Comment:
You mentioned "There's no Calc node at normalize plans, so ReduceExpressionRule.CALC does not match any thing.". So can we remove it from `DATASTREAM_NORM_RULES` and `DATASET_NORM_RULES`?

Reply:
yes, it can be removed from normalize phase